website/docs/reference/runtimes/temporal/index.mdx

```diff
@@ -1,13 +1,82 @@
+
+import SDKTabs from "@site/src/components/SDKTabs";
+import TabItem from "@theme/TabItem";
+
 # Temporal
 
+## Temporal runtime
+
 [Temporal](https://temporal.io/) is an open-source durable execution engine that can be used to develop workflows that are long lived and failure resistant.
-Common usecases include:
+Common use cases include:
 
 - Implementing multi-step, complicated transactionaly business logic. 
-- Guranteed event processing.
+- Guaranteed event processing.
 - Control planes for driving processes.
 
 The `TemporalRuntime` in Metatype can be used to directly `start`, `query`, `signal` and `describe` workflows on your temporal cluster. 
 Refer to the [temporal docs](https://docs.temporal.io) for more on what you can accomplish with this tech.
 
+An interesting use case is to dynamically describe the operations you want to expose, this enables reusing typegraphs accross different projects or even building a small framework around it.
+
+Here is a simple example of a typegraph that takes some value from an environment variable.
+
+<SDKTabs>
+    <TabItem value="python">
+```python
+from typegraph import t, typegraph, Policy, Graph
+from typegraph.providers.temporal import TemporalRuntime
+import os
```

Comment:
Can we externalize any valid source code so everything is testing and snapshotted?

Reply:
Will fix that in #676